components/brave_wallet/browser/solana_tx_manager_unittest.cc

```diff
@@ -429,6 +462,46 @@ class SolanaTxManagerUnitTest : public testing::Test {
     run_loop.Run();
   }
 
+  void TestRetryTransaction(const base::Location& location,
+                            const std::string& tx_meta_id,
+                            bool expected_result,
+                            const std::string& expected_err_message,
+                            const std::string& expected_blockhash = "") {
+    SCOPED_TRACE(testing::Message() << location.ToString());
+    MockTxStateManagerObserver observer(
+        solana_tx_manager()->GetSolanaTxStateManager());
+    if (expected_result) {
+      EXPECT_CALL(observer, OnNewUnapprovedTx(testing::_)).Times(1);
+    }
+
+    base::RunLoop run_loop;
+    solana_tx_manager()->RetryTransaction(
+        tx_meta_id,
+        base::BindLambdaForTesting([&](bool success, const std::string& id,
+                                       const std::string& err_message) {
+          EXPECT_EQ(expected_err_message, err_message);
+          EXPECT_EQ(expected_result, success);
+          if (expected_result) {
+            EXPECT_NE(tx_meta_id, id);
+            EXPECT_FALSE(id.empty());
+
+            // Get the new tx and check some fields.
+            auto tx_meta = solana_tx_manager()->GetTxForTesting(id);
```

Comment:
can we also check id != tx_meta_id?

Reply:
Yes, there's `EXPECT_NE(tx_meta_id, id);` above.